drivers/lps33hwtr.c

```diff
@@ -0,0 +1,235 @@
+    readData(reg, &r, 1);
+    return r;
+}
+
+
+static void lps33hwtr_reset(void) {
+    
+    ctx_t *ctx = &state;
+    ctx->ctrl1 = readReg(LPS35HW_CTRL_REG1);
+    // Reset and reboot
+    //BOOT|FIFO_EN|STOP_ON_FTH|IF_ADD_INC|I2C_DIS|[SWRESET]|0|ONE_SHOT
+    writeReg(LPS35HW_CTRL_REG2, LPS35HW_DEFAULT_CTRL_REG2 | 0x04); 
+    writeReg(LPS35HW_CTRL_REG1, ctx->ctrl1 ); 
+}
+
+static void lps33hwtr_set_rate(LPS35HW_DataRate new_rate)  {
+    ctx_t *ctx = &state;
+    //0|[ODR2]|[ODR1]|[ODR0]|EN_LPFP|LPFP_CFG|BDU|SIM
+    ctx->ctrl1 &= ~(0x70);                      //01110000
+    ctx->ctrl1 |= ((uint8_t)new_rate << 4);     //0XYZ0000 LSB
+
+    writeReg(LPS35HW_CTRL_REG1, ctx->ctrl1);
+}
+
+static void lps33hwtr_init(void) {
+    ctx_t *ctx = &state;
+    if (ctx->inited)
+        return;
+
+    ctx->pressure.min_value = SCALE_PRESSURE(260);
+    ctx->pressure.max_value = SCALE_PRESSURE(1260);
+    ctx->temperature.min_value = SCALE_TEMP(-40);
+    ctx->temperature.max_value = SCALE_TEMP(85);
+
+    ctx->nextsample = now;
+
+    ctx->inited = 1;
+
+    i2c_init();
+
+    int v = readReg(LPS35HW_WHO_AM_I);
+
+    DMESG("LPS35HW id: %x", v);
+
+    if (v == LPS35HW_ID) {
+        // OK
+        ctx->ctrl1 = readReg(LPS35HW_CTRL_REG1);
+        // Reset and reboot
+        //BOOT|FIFO_EN|STOP_ON_FTH|IF_ADD_INC|I2C_DIS|[SWRESET]|0|ONE_SHOT
+        writeReg(LPS35HW_CTRL_REG2, LPS35HW_DEFAULT_CTRL_REG2 | 0x04); 
+        writeReg(LPS35HW_CTRL_REG1, ctx->ctrl1 );
+    } else {
+        DMESG("invalid chip");
+        hw_panic();
+    }
+    lps33hwtr_set_rate(RATE_10_HZ);
+   //0|ODR2|ODR1|ODR0|EN_LPFP|LPFP_CFG|[BDU]|SIM
+   // setup block reads Default value: 0 (0: continuous update, 1:output registers not updated until MSB and LSB have been read)
+   ctx->ctrl1 = readReg(LPS35HW_CTRL_REG1);
+    writeReg(LPS35HW_CTRL_REG1, ctx->ctrl1 | 0x02); 
+}
+
+static void lps33hwtr_reset_pressure(void) {
+    uint8_t configInterrupt = readReg(LPS35HW_INTERRUPT_CFG);
+    configInterrupt &= ~(0x08);
+    configInterrupt |= ((uint8_t) 0x01 << 5);
+
+    writeReg(LPS35HW_INTERRUPT_CFG, configInterrupt);
+}
+
+
+static void lps33hwtr_set_pressure_zero(void) {
+    uint8_t configInterrupt = readReg(LPS35HW_INTERRUPT_CFG);
+    configInterrupt &= ~(0x08);
+    configInterrupt |= ((uint8_t) 0x01 << 4);
+
+    writeReg(LPS35HW_INTERRUPT_CFG, configInterrupt);
+}
+
+
+static uint32_t lps33hwtr_get_pressure(void) {
+    
+    uint32_t data[3];
+	readData(LPS35HW_PRESS_OUT_XL, (uint8_t *)data, 3);
+    uint32_t pressure = (data[2] << 16) | (data[1] << 8) | data[1];
+    pressure = pressure >>2;
+    
+    return pressure;
+}
+
+static int32_t lps33hwtr_get_temperature(void) {
+    int32_t temp;
+    int16_t rtemp = (readReg(LPS35HW_TEMP_OUT_H) << 8);
+    rtemp |= readReg(LPS35HW_TEMP_OUT_L);
+    temp = rtemp*10;
+    return temp;
+}
+
+static void lps33hwtr_process(void) {
+    ctx_t *ctx = &state;
+
+   if (jd_should_sample_delay(&ctx->nextsample, 1000)) {
```

Comment:
I think you can skip `read_issued`, and just use `jd_should_sample(&ctx->nextsample, SAMPLING_MS)` here; the `read_issued` is for sensors where you need to send a read command, wait a bit, and then read the results - see max31855.c for example

Reply:
Correct, I thought it might be useful for one-shot mode. I skipped  `read_issued`  for now as there's no way to change the mode anyway.